src/rgw/rgw_sal_motr.cc

```diff
@@ -3516,7 +3533,8 @@ int MotrStore::next_query_by_name(string idx_name,
                                   vector<bufferlist>& val_out,
                                   string prefix, string delim)
 {
-  unsigned nr_kvp = std::min(val_out.size(), 100UL);
+  // Caping the max-keys retreival to 1000 + 1 (extra) key.
+  unsigned nr_kvp = std::min(val_out.size(), 1001UL);
```

Comment:
I think this 1000 or 1001 value should be a constant but there is no specific class we have implemented like List in rados,
so don't really know where to declare this const.